src/game/player_callbacks_1.c

```diff
@@ -302,4 +325,491 @@ void PlayerCB_8025854(Player *player)
+                u16 dpadSideways = (player->unk5C & (DPAD_LEFT | DPAD_RIGHT));
+                if (dpadSideways != DPAD_RIGHT) {
+                    if (dpadSideways == DPAD_LEFT) {
+                        s32 val = player->speedGroundX;
+                        if (val <= 0) {
+                            player->moveState |= MOVESTATE_FACING_LEFT;
+                        } else if ((val - Q_24_8(0.09375)) < 0) {
+                            s32 deceleration = -Q_24_8(0.375);
+                            player->speedGroundX = deceleration;
+                        } else {
+                            player->speedGroundX = (val - Q_24_8(0.09375));
+                        }
+                    }
+                } else {
+                    s32 val = player->speedGroundX;
+                    if (val >= 0) {
+                        player->moveState &= ~MOVESTATE_FACING_LEFT;
+                    } else if ((val + Q_24_8(0.09375)) > 0) {
+                        player->speedGroundX = Q_24_8(0.375);
+                    } else {
+                        player->speedGroundX = (val + Q_24_8(0.09375));
+                    }
+                }
+            }
+        } else {
+            return;
+        }
+
+        if (player->speedGroundX > 0) {
+            player->unk50 = 8;
+        } else if (player->speedGroundX < 0) {
+            player->unk50 = -8;
+        } else {
+            player->unk50 = 0;
+        }
+
+        player->speedGroundX -= player->unk50;
+
+        if ((player->speedGroundX > -Q_24_8(0.5))
+            && (player->speedGroundX < Q_24_8(0.5))) {
+            player->unk50 = 0;
+            player->speedGroundX = 0;
+        }
+
+        if (player->speedGroundX == 0) {
+            gPlayer.callback = PlayerCB_8025318;
+            gPlayer.callback(player);
+        } else {
+            s32 speedX = player->speedGroundX;
+
+            if ((((player->rotation + Q_24_8(0.375)) & 0xFF) < 0xC0) && (speedX != 0)) {
+                u32 sinVal = SIN_24_8((player->rotation) * 4) * 60;
+                s32 sinInt = (s32)(Q_24_8_TO_INT((s32)sinVal));
+
+                if (speedX > 0) {
+                    if (sinInt <= 0) {
+                        sinInt >>= 2;
+                    }
+                } else if (sinInt >= 0) {
+                    sinInt >>= 2;
+                }
+
+                speedX += sinInt;
+
+                player->speedGroundX = speedX;
+            }
+
+            sub_80232D0(player);
+            sub_8023260(player);
+            sub_8023128(player);
+
+            if (player->moveState & MOVESTATE_IN_AIR) {
+                if (player->moveState & MOVESTATE_40) {
+                    player->speedAirY += Q_24_8(12.0 / 256.0);
+                } else {
+                    player->speedAirY += Q_24_8(42.0 / 256.0);
+                }
+            }
+
+            player->x += player->speedAirX;
+
+            if ((gUnknown_03005424 ^ gUnknown_0300544C)
```

Comment:
Probably need to make a macro for this

Reply:
Don't have to do now though

Reply:
Once we know WTF these variables are, sure! :D